src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,151 @@
+        user.setPassword("validePasssword");
+        assertThrows(InvalidUserDataException.class, () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    void registr_ageEdgeCase_notOk() {
+        user.setAge(17);
+        user.setLogin("validLogin");
+        user.setPassword("validePasssword");
+        assertThrows(InvalidUserDataException.class, () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    void registre_ageEquals18_ok() {
+        user.setAge(18);
+        user.setLogin("valLog");
+        user.setPassword("valPas");
+        User registeredUser = registrationServiceImpl.register(user);
+        assertEquals(user, registeredUser);
+    }
+
+    @Test
+    void registre_LogineExist_notOk() {
+        User existingUser = new User();
+        existingUser.setLogin("validLogin");
+        Storage.people.add(existingUser);
+        user.setAge(20);
+        user.setLogin("validLogin");
+        user.setPassword("validPass");
+        assertThrows(InvalidUserDataException.class, () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    void registre_LoginLength_notOk() {
+        user.setAge(20);
+        user.setLogin("log");
+        user.setPassword("validPass");
+        assertThrows(InvalidUserDataException.class, () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    void registre_LogineLengtthEgeCaseEquals15_notOk() {
+        user.setAge(20);
+        user.setLogin("valid");
+        user.setPassword("validPass");
+        assertThrows(InvalidUserDataException.class, () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    void register_LoginLengEquals_ok() {
+        user.setAge(18);
+        user.setLogin("validL");
+        user.setPassword("validP");
+        User registeredUser = registrationServiceImpl.register(user);
+        assertEquals(user, registeredUser);
+    }
+
+    @Test
+    void register_PasswordLength_notOk() {
+        user.setAge(20);
+        user.setLogin("validLogin");
+        user.setPassword("pass");
+        assertThrows(InvalidUserDataException.class, () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    void register_PasswordLength_Ok() {
+        user.setAge(18);
+        user.setLogin("validLogin");
+        user.setPassword("validP");
+        User registeredUser = registrationServiceImpl.register(user);
+        assertEquals(user, registeredUser);
+    }
+
+    @Test
+    void registr_PasswordLengtchEdgeCaseEqals_notOk() {
+        user.setAge(20);
+        user.setLogin("validLogin");
+        user.setPassword("valid");
+        assertThrows(InvalidUserDataException.class, () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    void registr_PasswordLengthEqalse_Ok() {
+        user.setAge(20);
+        user.setLogin("validLogin");
+        user.setPassword("validP");
+        assertThrows(InvalidUserDataException.class, () -> registrationServiceImpl.register(user));
```

Comment:
The test method `registr_PasswordLengthEqalse_Ok` contains typos and the test itself is incorrect because it asserts that an exception is thrown for a valid password length. Update the test to reflect the correct behavior.